### PR TITLE
[R20] CD-84824 xss_terminate now sanitized on assignment

### PR DESCRIPTION
**Pull request by johnny-lai**
- [x] @saghaulor 
- [x] @edk

### CODE REVIEW

**Comment by edk** [comment]
👍

**Comment by johnny-lai** [comment]
Allow `:if` option to be a Proc. The Proc is passed one argument, the record being sanitized.

**Comment by johnny-lai** [comment]
All xss_terminate_options can now be Procs. I also added a cache that is reset when the object is saved.

File: lib/xss_terminate/active_record.rb [hunk 1]
```diff
@@ -1,27 +1,38 @@
+require 'rails/all'
```
Comment (johnny-lai) [inline lib/xss_terminate/active_record.rb]: Replaced with `require 'active_record'` instead.

**Comment by johnny-lai** [comment]
Replace `:as` and `:if` options with `:format`

**Comment by saghaulor** [comment]
:shipit:

**Pull request merged**